### PR TITLE
[SPARK-37935][SQL] Eliminate separate error sub-classes fields

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to remove the error sub-class field from Spark exceptions, `checkError()` and from the machine-readable JSON format error message (`MINIMAL` and `STANDARD`, see the SQL config `spark.sql.error.messageFormat`), and include it to the original error class field using a dot. For example:

#### Spark exception

Before:
```scala
    new SparkRuntimeException(
      errorClass = "UNSUPPORTED_FEATURE",
      errorSubClass = "LITERAL_TYPE",
...
```

After:
```scala
    new SparkRuntimeException(
      errorClass = "UNSUPPORTED_FEATURE.LITERAL_TYPE",
...
```

#### JSON format

Before:
```json
{
  "errorClass" : "UNRESOLVED_COLUMN",
  "errorSubClass" : "WITH_SUGGESTION",
```

After:
```json
{
  "errorClass" : "UNRESOLVED_COLUMN.WITH_SUGGESTION",
```

### Why are the changes needed?
To simplify code and improve code maintenance.

### Does this PR introduce _any_ user-facing change?
Yes.

### How was this patch tested?
By running the affected test suites:
```
$ PYSPARK_PYTHON=python3 build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @itholic @srielau @anchovYu Could you review this PR, please.

**Review by cloud-fan — APPROVED** [review]

**Comment by srielau** [comment]
Uhm, shouldn't we discuss such design changes, before doing the work?Keep in mind that there is a lot more code dependent here than OSS.

How do you keep the nesting of the messages sane?

@MaxGekk @cloud-fan

**Comment by cloud-fan** [comment]
AFAIK we just combine the messages of the main and sub error classes. There should be no nesting?

File: core/src/main/scala/org/apache/spark/SparkThrowableHelper.scala [hunk 1]
```diff
@@ -34,29 +34,29 @@ private[spark] object SparkThrowableHelper {
   val errorReader = new ErrorClassesJsonReader(
     Seq(Utils.getSparkClassLoader.getResource("error/error-classes.json")))
 
+  def getErrorSubClass(errorClass: String): String = {
+    errorClass.split('.').lastOption.orNull
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply core/src/main/scala/org/apache/spark/SparkThrowableHelper.scala:38]: I am going to remove this method because it is not used.

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan @srielau for review.

**Pull request closed without merging**